docs/spec/concepts.rst

```diff
@@ -3,97 +3,190 @@
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if and only if ``B`` is a subtype of ``A``; or equivalently, if and only
+if the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on static types: the types ``A`` and
+``B`` are equivalent (or "the same type") if and only if ``A`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``. This means that the set of values
+represented by ``A`` is both a superset and a subset of the values represented
+by ``B``, meaning ``A`` and ``B`` must represent the same set of values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` (or as a "strict
+subtype" of ``A``) if ``B`` is a subtype of ``A`` and ``B`` is not equivalent
+to ``A``.
+
+The consistency relation
+------------------------
+
+Since :ref:`Any` represents an unknown static type, it does not represent any
+known single set of values, and thus it is not in the domain of the subtype,
+supertype, or equivalence relations on static types described above.
+
+We define a **materialization** relation on gradual types as follows: if
+replacing zero or more occurrences of ``Any`` in gradual type ``A`` with some
+gradual type (which can be different for each occurrence of ``Any``) results in
+the gradual type ``B``, then ``B`` is a materialization of ``A``. For instance,
+``tuple[int, str]`` (a static type) and ``tuple[Any, str]`` (a gradual type)
+are both materializations of ``tuple[Any, Any]``. ``tuple[int, str]`` is also a
+materialization of ``tuple[Any, str]``.
+
+If ``B`` is a materialization of ``A``, we can say that ``B`` is a "more
+static" type than ``A``, and ``A`` is a "more dynamic" type than ``B``.
+
+The materialization relation is both transitive and reflexive, so it defines a
+preorder on gradual types.
+
+We also define a **consistency** relation on gradual types.
+
+A static type ``A`` is consistent with another static type ``B`` if and only if
+they are the same type (``A`` is equivalent to ``B``.)
 
-Union types
------------
+A gradual type ``A`` is consistent with a gradual type ``B``, and ``B`` is
+consistent with ``A``, if and only if ``B`` is a materialization of ``A`` or
+``A`` is a materialization of ``B``.
+
+The dynamic type ``Any`` is consistent with every type, and every type is
+consistent with ``Any``. (This must follow from the above definitions of
+materialization and consistency, but is worth stating explicitly.)
+
+The consistency relation is not transitive. ``tuple[int, int]`` is consistent
+with ``tuple[Any, int]`` and ``tuple[Any, int]`` is consistent with
+``tuple[str, int]``, but ``tuple[int, int]`` is not consistent with
+``tuple[str, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
```

Comment:
Will we also mention that those concepts extend to programs? for example, if we want to talk about the gradual guarantee, then we want to talk about program A being more static than program B.

Reply:
I referenced this briefly in the section I added about the gradual guarantee.